packages/core/src/FrameLoop.ts

```diff
@@ -159,50 +172,53 @@ export class FrameLoop {
         const decay = config.decay === true ? 0.998 : config.decay
         position =
           from +
-          (velocity / (1 - decay)) *
-            (1 - Math.exp(-(1 - decay) * (time - startTime)))
+          (v0 / (1 - decay)) *
+            (1 - Math.exp(-(1 - decay) * animated.elapsedTime!))
+
+        // derivative of position
+        velocity = v0 * Math.exp(-(1 - decay) * animated.elapsedTime!)
 
         finished = Math.abs(animated.lastPosition - position) < 0.1
         if (finished) to = position
       }
       // Spring easing
       else {
-        let lastTime = animated.lastTime !== void 0 ? animated.lastTime : time
-        if (animated.lastVelocity !== void 0) {
-          velocity = animated.lastVelocity
-        }
+        velocity =
+          animated.lastVelocity !== void 0 ? animated.lastVelocity : v0 * 1000
```

Comment:
Why does `v0` need `* 1000`?

Reply:
This is an error from my end, but I realize velocity for spring is inconsistent between decay, easing and spring. I suspect spring velocity is expressed `px/seconds` while decay and easing is expressed in `px/milliseconds`, so there must be a need for harmonization there. In `react-use-gesture` velocity is expressed in milliseconds, I'll change it to whatever should be `react-spring` default.

@drcmda any advice?

Reply:
I've just pushed code to express velocity in `ms`.